frontend/src/components/TerminalGrid.tsx

```
@@ -0,0 +1,204 @@
+            <div className="add-agent-input">
+              <input
+                autoFocus
+                value={agentName}
+                onChange={(e) => setAgentName(e.target.value)}
+                onKeyDown={(e) => {
+                  if (e.key === "Enter") handleAddTerminal();
+                  if (e.key === "Escape") setAddingAgent(false);
+                }}
+                placeholder="Agent name..."
+              />
+              <button onClick={handleAddTerminal}>Add</button>
```

Biome (2.3.14)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)